tests/backup/backup_resiliency_test.go

```diff
@@ -758,3 +761,253 @@ var _ = Describe("{CancelAllRunningBackupJobs}", func() {
+			dash.VerifyFatal(originalReplicaCount > scaledDownReplica, true, "Verifying mongodb statefulset replica before taking backup")
+		})
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking backup of applications")
+			ctx, err = backup.GetAdminCtxFromSecret()
+			dash.VerifyFatal(err, nil, "Getting admin context")
+			for _, namespace := range appNamespaces {
+				for i := 0; i < numberOfBackups; i++ {
+					sem <- struct{}{}
+					backupName := fmt.Sprintf("%s-%s-%d", BackupNamePrefix, namespace, i)
+					backupNames = append(backupNames, backupName)
+					wg.Add(1)
+					go func(backupName string, namespace string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						defer func() { <-sem }()
+						_, err = CreateBackupWithoutCheck(backupName, SourceClusterName, bkpLocationName, backupLocationUID,
+							[]string{namespace}, labelSelectors, orgID, srcClusterUid, "", "", "", "", ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Taking backup %s of application- %s", backupName, namespace))
+					}(backupName, namespace)
+				}
+			}
+			wg.Wait()
+			log.Infof("The list of backups taken are: %v", backupNames)
+		})
+		Step("Scaling MongoDB statefulset replica to 0 and back to original replica while backup is in progress", func() {
+			log.InfoD("Scaling MongoDB statefulset replica to 0 while backup is in progress")
+			*statefulSet.Spec.Replicas = scaledDownReplica
+			statefulSet, err = apps.Instance().UpdateStatefulSet(statefulSet)
+			log.Infof("mongodb replica after scaling to 0 is %v", *statefulSet.Spec.Replicas)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Scaling down MongoDB statefulset replica to 0"))
+			dash.VerifyFatal(*statefulSet.Spec.Replicas == scaledDownReplica, true, "Verify mongodb statefulset replica after scaling down")
+			log.InfoD("Sleeping for 1 minute so that at least one request is hit to mongodb for the created backups")
+			time.Sleep(1 * time.Minute)
+			log.InfoD("Scaling MongoDB statefulset to original replica while backup is in progress")
+			statefulSet, err = apps.Instance().GetStatefulSet(mongodbStatefulset, backupNS)
+			*statefulSet.Spec.Replicas = originalReplicaCount
+			statefulSet, err = apps.Instance().UpdateStatefulSet(statefulSet)
+			log.Infof("mongodb replica after scaling back to original replica is %v", *statefulSet.Spec.Replicas)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Scaling MongoDB statefulset back to original replica"))
+			dash.VerifyFatal(*statefulSet.Spec.Replicas == originalReplicaCount, true, "Verify mongodb statefulset replica after scaling back to original")
+			log.Infof("Verify that at least one mongodb pod is in Ready state")
+			mongoDBPodStatus := func() (interface{}, bool, error) {
+				statefulSet, err = apps.Instance().GetStatefulSet(mongodbStatefulset, backupNS)
+				if err != nil {
+					return "", true, fmt.Errorf("unable to get the mongodb statefulset")
```

Comment:
To confirm, we don't want to retry getting mongoDBPodStatus again if we encounter an error, right?

Reply:
Wouldn't it be better to directly return the err in that case? This way, we would have a better understanding of the nature of the intermittent error instead of returning "unable to get the mongodb statefulset".

Reply:
corrected it